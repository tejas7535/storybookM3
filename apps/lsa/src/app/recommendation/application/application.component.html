<div
  class="flex flex-col gap-6 mb-8"
  *transloco="let t; read: 'recommendation.application'"
>
  <div class="py-8 flex flex-col gap-6">
    <span class="text-h6 text-medium-emphasis">
      {{ t('questionPowerType') }}</span
    >
    <lsa-radio-button-group
      [options]="powerSupplyRadioOptions"
      [control]="applicationForm.controls.battery"
    ></lsa-radio-button-group>
  </div>
</div>
