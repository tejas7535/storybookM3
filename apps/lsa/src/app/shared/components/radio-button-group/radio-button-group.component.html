<div class="flex flex-col md:gap-6 gap-2">
  <ng-container
    *ngIf="(isMediumScreen$ | ngrxPush) || simplifiedMobile; else mobileView"
  >
    <mat-radio-group
      [formControl]="control"
      class="flex flex-wrap"
      (change)="onOptionSelected($event)"
    >
      <div
        class="flex mb-4 md:mb-0"
        *ngFor="let option of options; let last = last"
        [matTooltip]="option.tooltip"
      >
        <mat-radio-button
          [value]="option.value"
          color="primary"
          [disabled]="option.disabled"
          class="cursor-pointer"
        >
          {{ option.name }}
        </mat-radio-button>
        <mat-divider [vertical]="true" *ngIf="!last" class="mx-4"></mat-divider>
      </div>
    </mat-radio-group>
    <div
      class="flex flex-row gap-4"
      *ngIf="visibleTemplate$ | ngrxPush as visibleTemplate"
    >
      <ng-template *ngTemplateOutlet="visibleTemplate.ref"></ng-template>
    </div>
  </ng-container>
</div>

<ng-template #mobileView>
  <ng-container *ngIf="!simplifiedMobile">
    <mat-radio-group
      [formControl]="control"
      class="flex flex-wrap"
      (change)="onOptionSelected($event)"
      ><div class="flex flex-col gap-2 w-full">
        <div
          *ngFor="let option of options"
          class="bg-secondary-variant flex flex-col px-2 py-4 rounded-lg gap-4"
        >
          <mat-radio-button
            [value]="option.value"
            color="primary"
            [disabled]="option.disabled"
            class="cursor-pointer"
          >
            {{ option.name }}
          </mat-radio-button>
          <div
            class="flex flex-col gap-2"
            *ngIf="
              option.value === control?.value &&
              templateMap[option.templateRef]?.ref
            "
          >
            <ng-template
              *ngTemplateOutlet="templateMap[option.templateRef].ref"
            ></ng-template>
          </div>
        </div></div></mat-radio-group
  ></ng-container>
</ng-template>
