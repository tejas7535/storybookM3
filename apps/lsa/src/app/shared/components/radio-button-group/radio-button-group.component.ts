import { BreakpointObserver } from '@angular/cdk/layout';
import { CommonModule } from '@angular/common';
import {
  AfterContentInit,
  ChangeDetectionStrategy,
  Component,
  ContentChildren,
  EventEmitter,
  Input,
  Output,
  QueryList,
  TemplateRef,
} from '@angular/core';
import { FormControl, ReactiveFormsModule } from '@angular/forms';
import { MatDividerModule } from '@angular/material/divider';
import { MatRadioChange, MatRadioModule } from '@angular/material/radio';
import { MatTooltipModule } from '@angular/material/tooltip';

import {
  distinctUntilChanged,
  map,
  Observable,
  shareReplay,
  startWith,
} from 'rxjs';

import { TAILWIND_SCREENS } from '@lsa/shared/constants';
import { PushPipe } from '@ngrx/component';

import { RadioOptionContentDirective } from './radio-option-content.directive';

@Component({
  selector: 'lsa-radio-button-group',
  standalone: true,
  imports: [
    CommonModule,
    MatRadioModule,
    ReactiveFormsModule,
    MatTooltipModule,
    MatDividerModule,
    PushPipe,
  ],
  templateUrl: './radio-button-group.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class RadioButtonGroupComponent<T> implements AfterContentInit {
  @Input() options: {
    value: T;
    name: string;
    disabled?: boolean;
    tooltip?: string;
    templateRef?: string;
  }[];
  @Input() control?: FormControl<T>;
  @Input() simplifiedMobile?: boolean;
  @Output() optionSelected = new EventEmitter<T>();

  @ContentChildren(RadioOptionContentDirective)
  templates!: QueryList<RadioOptionContentDirective>;

  isMediumScreen$: Observable<boolean> = this.breakpointObserver
    .observe([`(min-width: ${TAILWIND_SCREENS.MD})`])
    .pipe(map((state) => state.matches));

  templateMap: {
    [key: string]: {
      ref: TemplateRef<unknown>;
    };
  } = {};
  visibleTemplate$: Observable<
    typeof this.templateMap[keyof typeof this.templateMap] | undefined
  >;

  constructor(private readonly breakpointObserver: BreakpointObserver) {}

  ngAfterContentInit() {
    this.templateMap = {};
    this.templates.forEach((item) => {
      this.templateMap[item.name] = {
        ref: item.template,
      };
    });

    this.visibleTemplate$ = this.control.valueChanges.pipe(
      startWith(this.control.value),
      distinctUntilChanged(),
      shareReplay(1),
      map(
        (value) =>
          this.templateMap[
            this.options.find((option) => option.value === value)?.templateRef
          ]
      )
    );
  }

  onOptionSelected({ value }: MatRadioChange & { value: T }): void {
    this.control?.setValue(value);
    this.optionSelected.emit(value);
  }
}
