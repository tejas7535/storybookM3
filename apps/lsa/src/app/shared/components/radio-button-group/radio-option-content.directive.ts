import { Directive, Input, TemplateRef } from '@angular/core';

@Directive({
  // eslint-disable-next-line @angular-eslint/directive-selector
  selector: '[radioOptionContent]',
  // eslint-disable-next-line @angular-eslint/no-host-metadata-property
  host: {},
  standalone: true,
})
export class RadioOptionContentDirective {
  @Input('radioOptionContent') name: string | undefined;

  constructor(public template: TemplateRef<any>) {}
}
