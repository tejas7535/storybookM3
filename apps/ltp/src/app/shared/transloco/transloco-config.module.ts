import { APP_INITIALIZER, ModuleWithProviders, NgModule } from '@angular/core';

import {
  getBrowserLang,
  HashMap,
  TRANSLOCO_CONFIG,
  TranslocoConfig,
  TranslocoModule,
  TranslocoService
} from '@ngneat/transloco';

import { environment } from '../../../environments/environment';
import { translocoLoader } from './transloco.loader';

const translocoConfig: TranslocoConfig = {
  availableLangs: ['en', 'de'],
  defaultLang: 'en',
  fallbackLang: 'en',
  prodMode: environment.production,
  reRenderOnLangChange: true,
  flatten: {
    aot: environment.production
  }
};

const preloadLanguage = (transloco: TranslocoService): Function => {
  return async (): Promise<HashMap> => {
    const lang = getBrowserLang();

    transloco.setActiveLang(lang);

    return transloco.load(lang).toPromise();
  };
};

const preLoad = {
  provide: APP_INITIALIZER,
  multi: true,
  useFactory: preloadLanguage,
  deps: [TranslocoService]
};

@NgModule({
  imports: [TranslocoModule],
  exports: [TranslocoModule]
})
export class TranslocoConfigModule {
  static forRoot(): ModuleWithProviders {
    return {
      ngModule: TranslocoConfigModule,
      providers: [
        {
          provide: TRANSLOCO_CONFIG,
          useValue: translocoConfig
        },
        translocoLoader,
        preLoad
      ]
    };
  }
}
