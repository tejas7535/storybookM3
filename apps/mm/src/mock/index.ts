import { BearinxDialogDescription } from '@caeonline/dynamic-forms';

import { environment } from '../environments/environment';
import * as dialog from './dialog.json';

const envDialogString = JSON.stringify(dialog).replace(
  /OPTIONS_URL/g,
  environment.baseUrl
);

const envDialog = JSON.parse(envDialogString);

export const DIALOG = (envDialog as any).default as BearinxDialogDescription;
