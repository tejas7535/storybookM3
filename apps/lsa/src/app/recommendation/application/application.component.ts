import { CommonModule } from '@angular/common';
import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import { FormGroup, ReactiveFormsModule } from '@angular/forms';
import { MatDividerModule } from '@angular/material/divider';

import { RadioButtonGroupComponent } from '@lsa/shared/components/radio-button-group/radio-button-group.component';
import { PowerSupply } from '@lsa/shared/constants';
import { ApplicationForm } from '@lsa/shared/models';
import { translate, TranslocoModule } from '@ngneat/transloco';

const translatePath = 'recommendation.application';

@Component({
  selector: 'lsa-application',
  standalone: true,
  imports: [
    CommonModule,
    RadioButtonGroupComponent,
    TranslocoModule,
    MatDividerModule,
    ReactiveFormsModule,
  ],
  templateUrl: './application.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class ApplicationComponent {
  @Input()
  public applicationForm: FormGroup<ApplicationForm>;

  public readonly powerSupplyRadioOptions: {
    value: PowerSupply;
    name: string;
  }[] = [
    {
      value: PowerSupply.External,
      name: this.translateOption('external'),
    },
    {
      value: PowerSupply.Battery,
      name: this.translateOption('battery'),
    },
    {
      value: PowerSupply.NoPreference,
      name: this.translateOption('noPreference'),
    },
  ];

  private translateOption(option: string): string {
    const translationPath = `${translatePath}.powerOptions.${option}`;

    return translate(translationPath);
  }
}
