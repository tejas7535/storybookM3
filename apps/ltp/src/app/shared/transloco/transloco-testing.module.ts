import { TranslocoConfig, TranslocoTestingModule } from '@ngneat/transloco';

import * as en from '../../../assets/i18n/en.json';

export const getTranslocoModule = (config: Partial<TranslocoConfig> = {}) =>
  TranslocoTestingModule.withLangs(
    { en },
    {
      availableLangs: ['en'],
      defaultLang: 'en',
      ...config
    }
  );
