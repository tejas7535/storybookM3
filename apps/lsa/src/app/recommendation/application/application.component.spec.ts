import { RadioButtonGroupComponent } from '@lsa/shared/components/radio-button-group/radio-button-group.component';
import { mockApplicationForm } from '@lsa/testing/mocks/form.mock';
import { createComponentFactory, Spectator } from '@ngneat/spectator/jest';
import { TranslocoModule } from '@ngneat/transloco';

import { provideTranslocoTestingModule } from '@schaeffler/transloco/testing';

import { ApplicationComponent } from './application.component';

jest.mock('@ngneat/transloco', () => ({
  ...jest.requireActual<TranslocoModule>('@ngneat/transloco'),
  translate: jest.fn((translateKey) => translateKey),
}));

describe('ApplicationComponent', () => {
  let spectator: Spectator<ApplicationComponent>;
  let component: ApplicationComponent;

  const createComponent = createComponentFactory({
    component: ApplicationComponent,
    imports: [provideTranslocoTestingModule({ en: {} })],
    detectChanges: false,
  });

  beforeEach(() => {
    spectator = createComponent();
    component = spectator.debugElement.componentInstance;
    spectator.setInput('applicationForm', mockApplicationForm);
  });

  it('should create', () => {
    spectator.detectChanges();
    expect(component).toBeTruthy();
  });

  it('should have lsa-radio button component', () => {
    spectator.detectChanges();
    const radioButton = spectator.query(RadioButtonGroupComponent);
    expect(radioButton).toBeTruthy();
    expect(radioButton.options).toMatchSnapshot();
  });
});
