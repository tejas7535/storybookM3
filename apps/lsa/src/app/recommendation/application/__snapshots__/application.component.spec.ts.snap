// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationComponent should have lsa-radio button component 1`] = `
Array [
  Object {
    "name": "recommendation.application.powerOptions.external",
    "value": 0,
  },
  Object {
    "name": "recommendation.application.powerOptions.battery",
    "value": 1,
  },
  Object {
    "name": "recommendation.application.powerOptions.noPreference",
    "value": 2,
  },
]
`;
